driver.py

```diff
@@ -22,9 +24,27 @@
 class SmartSimScalingTests:
 
     def __init__(self):
-        self.resnet_model = "./imagenet/resnet50.pt"
         self.date = str(datetime.datetime.now().strftime("%Y-%m-%d"))
 
+    def set_resnet_model(self, device="GPU", force_rebuild=False):
+        self.resnet_model = f"./imagenet/resnet50.{device}.pt"
+        if not Path(self.resnet_model).exists() or force_rebuild:
+            logger.info(f"AI Model {self.resnet_model} does not exist or rebuild was asked, it will be created")
+            save_model(device=device)
```

Comment:
If save_model fails, it would be nice to generate an error and stop processing immediately

Reply:
Added a try-catch which exits immediately logging an error message!